lib/algo/manifold_interp/Interpolator.h

```diff
@@ -150,6 +127,39 @@ class Interpolator
         const Interpolator& rhs);
 };
 
+/**
+ * @brief Compute the RBF from the parameter points with the
+ *        unsampled parameter point.
+ *
+ * @param[in] parameter_points The parameter points.
+ * @param[in] interp_method  The interpolation method type ("LS" == linear solve,
+ *                           "IDW" == inverse distance weighting, "LP" == lagrangian polynomials)
+ * @param[in] rbf Which RBF to compute.
+ * @param[in] epsilon   The RBF parameter that determines the width of
+                        influence.
+ * @param[in] point The unsampled parameter point.
+ */
+std::vector<double> obtainRBFToTrainingPoints(std::vector<Vector*> parameter_points,
+    std::string interp_method, std::string rbf, double epsilon, Vector* point);
+
+/**
+ * @brief Compute the sum of the RBF weights.
+ *
+ * @param[in] rbf The vector holding the rbfs of the training points.
+ */
+double rbfWeightedSum(std::vector<double> rbf);
```

Comment:
Here and in some other places, `rbf` is passed by value. Is it certain to always be a small `vector`?